### ISSUE TITLE
1D Convolutional Neural Network not 1D as claimed

### ISSUE THREAD

**Issue by apinto-helomics** [MMostavi/CNNCancerType#3]
Hello,

I was hoping to mimic your method for the 1D convolutional neural network cited in your paper. When I came to assess your code for the 1D CNN I was surprised to see that all the code was a 2D CNN. Instead of inputting a vector as stated in the paper, it seems like whats inputted is the same expected input as the 2D CNN. Therefore I am curious if the code provided is incorrect? 
From your paper: 
"This CNN model takes the gene expression as a vector and applies one-dimensional kernels to the input vector [....] The input for 1D-CNN (Fig. 1a) is a 1D vector following gene symbol’s alphabetic order, " 

However per your code: 

     X_cancer_samples_mat = np.reshape(X_cancer_samples_mat, (-1, 71, 100))
    for train, test in kfold.split(X_cancer_samples_34, y_s):   # input_Xs in normal case and shuffled should be shuffled_Xs
    input_Xs = input_Xs.reshape(input_Xs.shape[0], img_rows, img_cols, 1)
    input_shape = (img_rows, img_cols, 1)
    input_Xs = input_Xs.astype('float32')

    num_classes = len(onehot_encoded[0])

    model = Sequential()
    ## *********** First layer Conv
    model.add(Conv2D(32, kernel_size=(1, 71), strides=(1, 1),
                     input_shape=input_shape))

You reshape the vector per the wrapper method into a 71 x 100 2D input. Is there other code doing what you claim in the paper?

Edit:
It looks like the code for the 1D CNN is actually the code for the 2D Hybrid CNN

**Comment by MMostavi** [comment]
Hi apinto-helomics,

I apologize for the late reply. Somehow I missed this question notification.

As you pointed out, I did not use 1D Convolution for the 1D-CNN model. Indeed, since the stride size and kernel size (which is 71) are same, I kept the 2D input and just slid 1*1 to have the 1D-CNN described in the model. Even if you reshape the input into vector and apply 1D convolution, the results would be same.
